### PR TITLE
[query] Generate better code in lowering rule for MatrixAggregate

### PR DESCRIPTION
**Pull request by tpoterba**
With StackStruct, now we don't allocate in the `MakeTuple` here.
Should deforest better, and be faster in general.

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala [hunk 1]
```diff
@@ -812,17 +813,20 @@ object LowerMatrixIR {
...
-              __entries_field = 'row (entriesField),
-              __cols_field = 'global (colsField)) {
-              irRange(0, '__entries_field.len)
-                .filter(idx ~> !'__entries_field (idx).isNA)
-                .aggExplode(idx ~> aggLet(sa = '__cols_field (idx), g = '__entries_field (idx)) {
-                  subst(query, matrixSubstEnv(child))
-                })
-            })
+        TableAggregate(lc,
+          aggExplodeIR(filterIR(
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala:817]: The indentation is making this hard to parse. Could you maybe start the `filterIR` on the next line and indent its arguments, and put the filter body on its own line?

**Review by patrick-schultz — CHANGES_REQUESTED** [review]

File: hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala:817]: oh, sure.

**Review by chrisvittal — APPROVED** [review]
